### PR TITLE
Add option to replace binary separator with pipe

### PR DESCRIPTION
**Pull request by Nadia-Adaptive**
A change to FixArchivePrinter to allow replacing the binary separator in the FIX message body with a pipe via program arguments.

### CODE REVIEW

**Comment by vyazelenko** [comment]
@Nadia-Adaptive Have you considered adding `--delimiter=X` so that any character can be specified?

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/logger/FixArchivePrinter.java [hunk 1]
```diff
@@ -387,8 +405,8 @@ private static void print(
     {
         final MessageStatus status = message.status();
         final long timestamp = message.timestamp();
-        final String body = message.body();
-        System.out.printf("%1$20s: %2$s (%3$s)%n", timestamp, body, status);
+        final String body = pipeDelimiter ? message.body().replace('\u0001', '|') : message.body();
```
Comment (chrjohn) [inline artio-core/src/main/java/uk/co/real_logic/artio/engine/logger/FixArchivePrinter.java:408]: I don't know how far you would like to take this, but `SOH` could also appear in `data` fields, so you might end up with pipe symbols in fields that should not be replaced.
But I think this approach is good enough for the majority of cases.

**Comment by wojciech-adaptive** [comment]
> @Nadia-Adaptive Have you considered adding `--delimiter=X` so that any character can be specified?

We should implement this.

File: artio-core/src/main/java/uk/co/real_logic/artio/engine/logger/FixArchivePrinter.java (hunk 1 above)
Comment (wojciech-adaptive) [reply artio-core/src/main/java/uk/co/real_logic/artio/engine/logger/FixArchivePrinter.java:408]: Good point, thanks. I think this is good enough for now, `fix.core.debug.separator` doesn't handle data fields either.

File: artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/ArchivePrinterIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (wojciech-adaptive) [inline artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/ArchivePrinterIntegrationTest.java:126]: formatting

File: artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/ArchivePrinterIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import static uk.co.real_logic.artio.system_tests.SystemTestUtil.connect;
+import static uk.co.real_logic.artio.system_tests.SystemTestUtil.delete;
+import static uk.co.real_logic.artio.system_tests.SystemTestUtil.launchInitiatingEngine;
+import static uk.co.real_logic.artio.system_tests.SystemTestUtil.newInitiatingLibrary;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import org.junit.Before;
+import org.junit.Test;
```
Comment (wojciech-adaptive) [inline artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/ArchivePrinterIntegrationTest.java:33]: Let's use JUnit 5 for new tests.

Comment (wojciech-adaptive) [reply artio-system-tests/src/test/java/uk/co/real_logic/artio/system_tests/ArchivePrinterIntegrationTest.java:33]: Actually we probably can't due to the base class.

**Pull request merged**